scripts/o3de/o3de/utils.py

```diff
@@ -346,3 +365,97 @@ def remove_gem_duplicates(gems: list) -> list:
+        return dict(entry.split('=') for entry in replace_values if '=' in entry)
+    
+    if new_values:
+        new_values = new_values.split() if isinstance(new_values, str) else new_values
+        new_values = dict(entry.split('=') for entry in new_values if '=' in entry)
+        if new_values:
+            existing_values.update(new_values)
+    
+    if remove_values:
+        remove_values = remove_values.split() if isinstance(remove_values, str) else remove_values
+        [existing_values.pop(key) for key in remove_values]
+    
+    return existing_values
+
+def update_values_in_key_list(existing_values: list, new_values: list or str, remove_values: list or str,
+                      replace_values: list or str):
+    """
+    Updates values within a list by replacing all values or by appending values in the new_values list, 
+    removing values in the remove_values and then removing duplicates.
+    :param existing_values list with existing values to modify
+    :param new_values list with values to add to the existing value list
+    :param remove_values list with values to remove from the existing value list
+    :param replace_values list with values to replace in the existing value list
+
+    returns updated existing value list
+    """
+    if replace_values != None:
+        replace_values = replace_values.split() if isinstance(replace_values, str) else replace_values
+        return list(dict.fromkeys(replace_values))
+
+    if new_values:
+        new_values = new_values.split() if isinstance(new_values, str) else new_values
+        existing_values.extend(new_values)
+    if remove_values:
+        remove_values = remove_values.split() if isinstance(remove_values, str) else remove_values
+        existing_values = list(filter(lambda value: value not in remove_values, existing_values))
+
+    # replace duplicate values
+    return list(dict.fromkeys(existing_values))
+
+
+def get_object_name_and_version_specifier(input:str) -> (str, str) or None:
+    """
+    Get the object name and version specifier from a string in the form <name><version specifier(s)>
+    Valid input examples:
+        o3de>=1.2.3
+        o3de-sdk==1.2.3,~=2.3.4
+    :param input the string in the form <name><version specifier(s)>
+
+    return an engine name and a version specifier or raises an exception if input is invalid
+    """
+
+    regex_str = r"(?P<object_name>(.*?))(?P<version_specifier>((~=|==|!=|<=|>=|<|>|===)(\s*\S+)+))"
+    regex = re.compile(r"^\s*" + regex_str + r"\s*$", re.VERBOSE | re.IGNORECASE)
+    match = regex.search(input)
+
+    if not match:
+        raise Exception(f"Invalid name and/or version specifier {input}, expected <name><version specifiers> e.g. o3de==1.2.3")
+
+    if not match.group("object_name"):
+        raise Exception(f"Invalid or missing name {input}, expected <name><version specifiers> e.g. o3de==1.2.3")
```

Comment:
We should make an specific exception class for this.
We should narrow our exception specifications to only catch the exceptions we expect.
The ones that aren't expected we want to surface loudly so that they can be fixed.
For example if the regex compile function raised an exception after an update.